### PR TITLE
More work/cleanup for the serial example

### PR DESCRIPTION
**Pull request by Ivan-Velickovic**
* Fixes/cleanup in Makefile
* Fixes in PL011 UART driver
* Get the example working on QEMU since we have the UART driver for it.

### CODE REVIEW

File: examples/serial/README.md [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
+If you wish to simulate on the QEMU ARM virt platform, you can run the following command:
+```sh
+make MICROKIT_SDK=<path/to/sdk> MICROKIT_BOARD=qemu_arm_virt qemu
 ```
 
 ## Running/using
 
-To run the image, load the `loader.img` (located in the build directory) binary onto an Odroid-C4.
-Upon loading the image, the following will be outputted to the terminal:
+Upon booting the system image, the following will be outputted to the terminal:
```
Comment (wom-bat) [inline examples/serial/README.md:24]: 'outputted' is a horrible word.  'written ' maybe?

File: examples/serial/board/qemu_arm_virt/serial.system [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    <memory_region name="tx_data_driver" size="0x200_000" page_size="0x200_000" />
+    <memory_region name="tx_data_client" size="0x200_000" page_size="0x200_000" />
+    <memory_region name="rx_data_driver" size="0x200_000" page_size="0x200_000" />
+    <memory_region name="rx_data_client" size="0x200_000" page_size="0x200_000" />
+
+    <memory_region name="tx_data_client2" size="0x200_000" page_size="0x200_000" />
+    <memory_region name="rx_data_client2" size="0x200_000" page_size="0x200_000" />
+
+    <!-- shared memory for ring buffer mechanism : serial -->
+    <memory_region name="rx_free_serial_driver" size="0x200_000" page_size="0x200_000"/>
```
Comment (wom-bat) [inline examples/serial/board/qemu_arm_virt/serial.system:14]: Pretty sure these can be smaller than 1M.

File: examples/serial/board/qemu_arm_virt/serial.system [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        <map mr="rx_free_serial_driver" vaddr="0x4_000_000" perms="rw" cached="true" setvar_vaddr="rx_free" />
+        <map mr="rx_active_serial_driver" vaddr="0x4_200_000" perms="rw" cached="true" setvar_vaddr="rx_active" />
+        <map mr="tx_free_serial_driver" vaddr="0x4_400_000" perms="rw" cached="true" setvar_vaddr="tx_free" />
+        <map mr="tx_active_serial_driver" vaddr="0x4_600_000" perms="rw" cached="true" setvar_vaddr="tx_active" />
+
+        <map mr="tx_data_driver" vaddr="0x2_200_000" perms="rw" cached="true" />
+        <map mr="rx_data_driver" vaddr="0x2_600_000" perms="rw" cached="true" />
+
+        <!-- UART interrupt -->
+        <irq irq="33" id="1" />
```
Comment (wom-bat) [inline examples/serial/board/qemu_arm_virt/serial.system:44]: This IRQ is for the QEMU.  Can it be parameterised here to use the 255,edge for Meson or 33.level for QEMU?

Comment (wom-bat) [reply examples/serial/board/qemu_arm_virt/serial.system:44]: So you can use the same system.xml file for both boards?

**Review by wom-bat — CHANGES_REQUESTED** [review]
Needs updating to be able to merge.  There are conflicts.

File: examples/serial/README.md (hunk 1 above)
Comment (Ivan-Velickovic) [reply examples/serial/README.md:24]: Yes we should change that.

File: examples/serial/board/qemu_arm_virt/serial.system (hunk 2 above)
Comment (Ivan-Velickovic) [reply examples/serial/board/qemu_arm_virt/serial.system:14]: Will make an issue and do it in a separate PR.

File: examples/serial/board/qemu_arm_virt/serial.system (hunk 3 above)
Comment (Ivan-Velickovic) [reply examples/serial/board/qemu_arm_virt/serial.system:44]: I don't want to do any extra indirection on the examples, it makes sense for LionsOS but not for these isolated examples.

File: examples/serial/README.md (hunk 1 above)
Comment (Ivan-Velickovic) [reply examples/serial/README.md]: Changed to "Upon booting the system image, you will see the following output:"

File: examples/serial/board/qemu_arm_virt/serial.system (hunk 2 above)
Comment (Ivan-Velickovic) [reply examples/serial/board/qemu_arm_virt/serial.system:14]: [Done](https://github.com/au-ts/sDDF/issues/66)

**Review by wom-bat — APPROVED** [review]
LGTM

**Pull request merged**